server/server.js

```diff
@@ -1,15 +1,99 @@
 const express = require("express");
 const app = express();
-const port = process.env.PORT || 5000;
+app.use(express.json());
+const port = process.env.PORT || 5001;
+const fs = require("fs");
+const cors = require ("cors")
+app.use(cors)
 
 app.listen(port, () => console.log(`Listening on port ${port}`));
 
-// Store and retrieve your videos from here
-// If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+let videos = JSON.parse(fs.readFileSync("videos.json", "utf-8"));
+
+const save = () => {
+  fs.writeFileSync("videos.json", JSON.stringify(videos, null, 2));
+};
+
+function matchYoutubeUrl(url) {
+  var p =
+    /^(?:https?:\/\/)?(?:m\.|www\.)?(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((\w|-){11})(?:\S+)?$/;
+  if (url.match(p)) {
+    return url.match(p)[1];
+  }
+  return false;
+}
 
 // GET "/"
+
 app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+  res.json(videos);
+});
+
+app.get("/:id", (req, res) => {
+  const id = parseInt(req.params.id);
+  if (isNaN(id)) {
+    res.sendStatus(400);
+    return;
+  }
+  const findVideo = videos.find((a) => a.id === id);
+
+  if (!findVideo) {
+    res.sendStatus(404);
+    return;
+  }
+  res.send(findVideo);
+});
+
+// POST
+
+const compulsoryFields = ["title", "url", "rating"];
+
+app.post("/", (req, res) => {
+  if (!compulsoryFields.every((cf) => req.body.hasOwnProperty(cf))) {
```

Comment:
Although group validation would work just fine, breaking it down and validating the title, the url and the rating separately makes your server response more specific and less prone to errors.